### PR TITLE
Remove an unused initialization argument from `Audit::CatalogToMoab`

### PR DESCRIPTION
**Pull request by mjgiarlo**
## Why was this change made?

I am not sure why this argument was there and not used when the storage location can be grabbed from the `CompleteMoab`. Possibly used to be used for dependency injection or testing? Instead, grab the storage location from the `CompleteMoab` since we no longer appear to be using this arg which is mostly getting in the way.

## Was the usage documentation (e.g. wiki, README, queue or DB specific README) updated?

no

### CODE REVIEW

File: app/lib/audit/catalog_to_moab.rb [hunk 1]
```diff
@@ -45,10 +44,12 @@ def check_catalog_version
       compare_version_and_take_action
     end
 
-    alias storage_location storage_dir
-
     private
 
+    def storage_location
+      complete_moab.moab_storage_root.storage_location
```
Comment (ndushay) [inline app/lib/audit/catalog_to_moab.rb]: is there any advantage in memoizing this?

also - we're sure this won't cause multiple "which storage_root am i on?" queries (seems like it won't) ... cause those can be expensive.

Comment (mjgiarlo) [reply app/lib/audit/catalog_to_moab.rb]: I... don't know! What do you think, @jmartin-sul?

Comment (jmartin-sul) [reply app/lib/audit/catalog_to_moab.rb]: my recollection is that AR will essentially do some naive caching for us.  there'll be a query for the `#moab_storage_root` call, but it turns out we're already doing that in the constructor to get an `AuditResults` object:  https://github.com/sul-dlss/preservation_catalog/blob/5398420867e2c91f4dc594643dfc698500bf9fbd/app/lib/audit/catalog_to_moab.rb#L9-L13

and then i believe we'll be accessing the same instance of `MoabStorageRoot` until we call `#reload` on either `complete_moab` or `complete_moab.moab_storage_root`.

basically my recollection is that ActiveRecord doesn't re-query for accessing a field once the data has been retrieved and the object constructed.

File: app/lib/audit/catalog_to_moab.rb [hunk 2]
```diff
@@ -45,10 +44,12 @@ def check_catalog_version
...
+    end
```
Comment (jmartin-sul) [inline app/lib/audit/catalog_to_moab.rb]: hmm, so actually, nothing even appears to call `storage_location` anymore.  i think we could get rid of the method entirely.  nothing in this class seems to refer to it, afaict, and grepping through the codebase, only one thing seems to instantiate this class:
https://github.com/sul-dlss/preservation_catalog/blob/5398420867e2c91f4dc594643dfc698500bf9fbd/app/jobs/catalog_to_moab_job.rb#L14-L16

and nothing there references `storage_location` .

File: spec/lib/audit/catalog_to_moab_spec.rb [hunk 3]
```diff
@@ -71,7 +70,7 @@
       c2m.check_catalog_version
     end
 
-    it 'calls online_moab_found?(druid, storage_dir)' do
+    it 'calls online_moab_found?(druid, storage_location)' do
```
Comment (jmartin-sul) [inline spec/lib/audit/catalog_to_moab_spec.rb]: looking through the code, i'm sort of guessing this parenthetical is cruft from a prior method signature.  or at least, i don't really understand what this parenthetical is communicating at present.  will leave a suggestion that gets rid of it, and you can squash it in (or not) if you'd like.

Comment (jmartin-sul) [inline spec/lib/audit/catalog_to_moab_spec.rb]:
```suggestion
    it 'calls online_moab_found?' do
```

**Comment by mjgiarlo** [comment]
@jmartin-sul :speech_balloon: 
> ~assuming it's green, feel free to squash into yours and merge this PR if you agree that my follow-on makes sense.~ _retracting that PR, missed where it was called due to some indirection i didn't notice, sorry!_

Yep, `MoabValidationHandler`, right? It's that whole thing where you write a module that expects certain messages but there's no good way to enforce that contract. (This is what Joe used to say "modules not being modular.")

**Comment by jmartin-sul** [comment]
> @jmartin-sul 💬
> 
> > ~assuming it's green, feel free to squash into yours and merge this PR if you agree that my follow-on makes sense.~ _retracting that PR, missed where it was called due to some indirection i didn't notice, sorry!_
> 
> Yep, `MoabValidationHandler`, right? It's that whole thing where you write a module that expects certain messages but there's no good way to enforce that contract. (This is what Joe used to say "modules not being modular.")

indeed, and that module was my idea iirc, so got what i deserved 😆

File: app/lib/audit/catalog_to_moab.rb (hunk 1 above)
Comment (mjgiarlo) [reply app/lib/audit/catalog_to_moab.rb]: @ndushay What do you think? :point_up:

File: spec/lib/audit/catalog_to_moab_spec.rb (hunk 3 above)
Comment (mjgiarlo) [reply spec/lib/audit/catalog_to_moab_spec.rb]: Committed!

File: app/lib/audit/catalog_to_moab.rb (hunk 1 above)
Comment (mjgiarlo) [reply app/lib/audit/catalog_to_moab.rb]: @ndushay What say you re: :point_up: ?

Comment (ndushay) [reply app/lib/audit/catalog_to_moab.rb]: all good.  Go for it. Push the green button.
